src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/resource/ReviewScheduleEtlController.kt

```diff
@@ -0,0 +1,174 @@
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.ResponseStatus
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.domain.learningandworkprogress.review.SentenceType
+import uk.gov.justice.digital.hmpps.domain.learningandworkprogress.review.dto.CreateInitialReviewScheduleDto
+import uk.gov.justice.digital.hmpps.domain.learningandworkprogress.review.service.ReviewService
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.client.prisonersearch.LegalStatus
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.database.jpa.repository.ActionPlanRepository
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.database.jpa.repository.InductionRepository
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.database.jpa.repository.ReviewScheduleRepository
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.service.PrisonerSearchApiService
+
+private val log = KotlinLogging.logger {}
+
+/**
+ * Temporary controller class exposing REST API method that triggers the ETL of ReviewSchedule records
+ */
+@Hidden
+@RestController
+class ReviewScheduleEtlController(
+  private val prisonerSearchApiService: PrisonerSearchApiService,
+  private val inductionRepository: InductionRepository,
+  private val actionPlanRepository: ActionPlanRepository,
+  private val reviewScheduleRepository: ReviewScheduleRepository,
+  private val reviewService: ReviewService,
+) {
+
+  @PostMapping
+  @ResponseStatus(HttpStatus.CREATED)
+  @PreAuthorize(HAS_EDIT_REVIEWS)
+  @RequestMapping(value = ["/action-plans/review-schedules/etl/{prisonId}"])
+  fun createReviewSchedulesForPrisonersInPrison(@PathVariable("prisonId") prisonId: String): ReviewSchedulesEtlResponse {
+    // Create a ReviewSchedule for all prisoners in the given prison who need one
+
+    val totalPrisonersInPrison: Int
+    val totalPrisonersWithReviewSchedule: Int
+    val totalPrisonersWithInduction: Int
+    val totalPrisonersWithActionPlan: Int
+    val totalSentencedPrisonersWithNoReleaseDate: Int
+    val prisonersWithCreatedReviewSchedules = mutableListOf<String>()
+    val prisonersWithoutReviewSchedules = mutableListOf<String>()
+
+    // Get all prisoners in the given prison
+    prisonerSearchApiService.getAllPrisonersInPrison(prisonId)
+      .also { totalPrisonersInPrison = it.size }
+      // Filter the prisoners, keeping only those who do not already have a Review Schedule, and who have both an Induction and Action Plan with at least 1 goal
+      // Get those without any Review Schedule (most DB efficient is a direct call to the repository findByPrisonNumberIn method)
+      .let { prisoners ->
+        val prisonNumbers = prisoners.map { it.prisonerNumber }
+        val prisonersWithReviewSchedule = reviewScheduleRepository.findByPrisonNumberIn(prisonNumbers)
```

Comment:
Not sure if this will be a problem or not but if there are 1000s of prisoners will the IN sql work? 
according to [this](https://stackoverflow.com/questions/1009706/postgresql-max-number-of-parameters-in-in-clause#:~:text=At%20least%20I%20can%20tell,using%20Posgresql's%20JDBC%20driver%209.1.) the limit is 65000 bytes: so with the prisoner number being 7 bytes (assuming UTF8) then that is a limit of just under 10k prisoner numbers... assuming the stack over flow link is correct.

Reply:
Hmmm, good question
I wonder if we can do a bit of analysis to understand the highest occupancy of all prisons ....

Bearing in mind this will be run per-prison, it _might_ be OK ?